app/javascript/js/controllers/dashboard_card_controller.js

```diff
@@ -1,9 +1,24 @@
 import { Controller } from '@hotwired/stimulus'
+import URI from 'urijs'
+
+function updateQueryParam(url, key, value) {
+  const uri = new URI(url)
+  uri.setQuery(key, value)
+
+  return uri.toString()
+}
+
+function getDateParam(url) {
+  const queryString = url.split('?')[1]
```

Comment:
Why are we splitting this as a string?
Can't we use `URI` as well to pick up that param?

Please let's do that.

Reply:
Yes, sure! I will update it using URI.